File: Rust/selection_sort.rs
// Selection sort

fn sort(arr: &mut [i32]) {
    let len = arr.len();
    for i in 0..len {
        let mut temp = i;
        for j in (i+1)..len {
            if arr[temp] > arr[j] {
                temp = j;
            }
        }
        arr.swap(i, temp);
    }
}


fn main () {
    let mut test_arr:[i32; 7] = [7, 4, 1, 2, 0, 9, 6];
    println!("{:?}", test_arr);
    sort(&mut test_arr);
    println!("{:?}", test_arr);
}

File: C#/selection_sort.cs
// C# program for Selection Sort
using System;

class Program
{
	static void sort(int []arr)
	{
		int n = arr.Length;
    
		for (int i = 0; i < n - 1; i++)
		{
			int indx = i;
			for (int j = i + 1; j < n; j++){
				if (arr[j] < arr[indx]){
					indx = j;
				}
			}

			// Swap the found minimum element with the first element
			int temp = arr[indx];
			arr[indx] = arr[i];
			arr[i] = temp;
		}
	}

	// Prints the array
	static void printArray(int []arr)
	{
		int n = arr.Length;
		for (int i=0; i<n; ++i){
			Console.Write(arr[i]+" ");
		}
		Console.WriteLine();
	}

	// Test code
	public static void Main()
	{
		int []arr = {55,44,33,22,11};
		sort(arr);
		Console.WriteLine("Sorted array:");
		printArray(arr);
	}

}
